### PR TITLE
Configure Travis

### PR DESCRIPTION
**Pull request by mlautman**
### Description

This pr attempts to make the necessary changes to get the master branch to pass travis. 

It depends on #56 and #46 

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Document API changes relevant to the user in the moveit/MIGRATION.md notes
- [ ] Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Decide if this should be cherry-picked to other current ROS branches
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,28 +14,18 @@ notifications:
...
     - MOVEIT_CI_TRAVIS_TIMEOUT=85  # Travis grants us 90 min, but we add a safety margin of 5 min
-    - ROS_DISTRO=melodic
+    - ROS_DISTRO=crystal
     - ROS_REPO=ros
     - UPSTREAM_WORKSPACE=moveit.rosinstall
     - TEST_BLACKLIST=moveit_ros_perception  # mesh_filter_test fails due to broken Mesa OpenGL
     - CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-unused-parameter -Wno-unused-but-set-parameter -Wno-unused-function"
     - WARNINGS_OK=false
   matrix:
-    - TEST="clang-format catkin_lint"
```
Comment (davetcoleman) [inline .travis.yml]: @rhaschke added an ament_lint, let's keep that enabled or at least commented out with TODO so we remember to add it later

File: .travis.yml [hunk 2]
```diff
@@ -14,28 +14,18 @@ notifications:
...
-    - TEST=clang-tidy-fix
```
Comment (davetcoleman) [inline .travis.yml]: add TODO to re-enable in future

File: .travis.yml [hunk 3]
```diff
@@ -14,28 +14,18 @@ notifications:
...
-    - ROS_DISTRO=melodic
-    - ROS_DISTRO=kinetic
-
-matrix: # Add a separate config to the matrix, using clang as compiler
-  include:
-    # add a config to the matrix using clang as compiler and also test ikfast plugin creation
-    - compiler: clang
-      env: ROS_REPO=ros-shadow-fixed
-           BEFORE_DOCKER_SCRIPT="source moveit_kinematics/test/test_ikfast_plugins.sh"
-           CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-unused-parameter -Wno-unused-function -Wno-overloaded-virtual"
```
Comment (davetcoleman) [inline .travis.yml]: This is some pretty fancy ikfast testing stuff @rhaschke did, at least comment out with TODO to re-instate when ready

**Comment by davetcoleman** [comment]
In your PR description, please add more detail about what you did i.e. "ran clang-format on whole codebase"

better: put clang-formatting in different PR

File: .travis.yml (hunk 3 above)
Comment (mlautman) [reply .travis.yml]: I believe @rhaschke was the one who first suggested we remove it here. We always have the moveit repo to go back to for examples.

Comment (rhaschke) [reply .travis.yml]: There is no reason to remove this. It should compile for MoveIt2 as well, shouldn't it?

File: .travis.yml (hunk 1 above)
Comment (mlautman) [reply .travis.yml]: done

File: .travis.yml (hunk 2 above)
Comment (mlautman) [reply .travis.yml]: done

File: .travis.yml (hunk 3 above)
Comment (mlautman) [reply .travis.yml]: Restored for now. If it doesn't pass I'll comment out with a todo

**Comment by mlautman** [comment]
`panda_ikfast_plugin: Cannot locate rosdep definition for [eigen_conversions]`

File: .travis.yml (hunk 3 above)
Comment (mlautman) [reply .travis.yml]: `panda_ikfast_plugin: Cannot locate rosdep definition for [eigen_conversions]`

Comment (rhaschke) [reply .travis.yml]: If you already figured out that eigen_conversions is not available in ROS2 anymore, why not directly replace it with some tf2 equivalents?

**Comment by rhaschke** [comment]
Right now, Travis just tests that 3rd-party dependencies in the ROS_WS compile.
All MoveIt stuff is disabled by COLCON_IGNORE.

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # This config file for Travis CI utilizes https://github.com/ros-planning/moveit_ci/ package.
 sudo: required
-dist: xenial  # distro used by Travis, moveit_ci uses the docker image's distro
+dist: trusty  # distro used by Travis, moveit_ci uses the docker image's distro
```
Comment (henningkayser) [inline .travis.yml]: Why should Travis tun on trusty? Shouldn't we use bionic?

Comment (gavanderhoorn) [reply .travis.yml]: Travis only provides two distributions: `trusty` and `xenial` (and the latter was only added relatively recently).

Comment (davetcoleman) [reply .travis.yml]: I even added an in-line comment for this, above! I guess it's not clear enough.
https://github.com/ros-planning/moveit2/pull/77

**Pull request merged**